### PR TITLE
Feature/dssm preprocessor

### PR DESCRIPTION
**Pull request by bwanglzu**
1. add inputs argument in `base_processor`.
2. Add `DSSMPreprocessor` class as a helper function.

For issue #245 `DSSMPreprocessor` severe as  facade class of varies `ProceesorUnits` to pre-process inputs for DSSM model.

### CODE REVIEW

File: matchzoo/preprocessor/dssm_preprocessor.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        output_left = []
+        output_righ = []
+        if not self._context.get('term_index'):
+            raise ValueError(
+                "Please fit term_index before apply transofm function")
+        inputs, labels = self._detach_labels(inputs)
+        units = self._prepare_stateless_units()
+        units.append(
+            preprocessor.WordHashingUnit(self._context['term_index']))
+        for left, righ in inputs:
```
Comment (faneshion) [inline matchzoo/preprocessor/dssm_preprocessor.py]: why not use 'right' in `for left, righ in inputs:`.

File: matchzoo/engine/base_preprocessor.py [hunk 2]
```diff
@@ -18,17 +18,55 @@ def fit(self) -> 'BasePreprocessor':
...
+        the inputs should be list of tuples like [(query, document)].
+        This method is used to infer the stage, if `label` exist, detach
+        label to a list.
+
+        :param inputs: List of text-left, text-right, label triples.
+        :return: Zipped list of text-left and text-right.
+        :return: Detached labels, list or None.
+        """
+        unzipped_inputs = list(zip(*inputs))
+        if len(unzipped_inputs) == 3:
```
Comment (faneshion) [inline matchzoo/engine/base_preprocessor.py]: For common usage, I think 'text id' is often useful to determine the document belongs to one specific query. In this case, the input should be like `[{'text_left': [], 'text_right': [], 'label': 0, 'ids': ('qid0', 'did1')}, {'text_left': [], 'text_right': [], 'label': 1, 'ids': [('qid0', 'did2')]}] ....`.

File: matchzoo/preprocessor/dssm_preprocessor.py (hunk 1 above)
Comment (bwanglzu) [reply matchzoo/preprocessor/dssm_preprocessor.py]: @faneshion 我的强迫症让我想要把它们对齐了..

**Comment by faneshion** [comment]
I would approve this branch to make the dssm can be test.

**Pull request merged**